### PR TITLE
Added folding for type and var declarations and composite literals

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

File: src/com/goide/editor/GoFoldingBuilder.java [hunk 1]
```diff
@@ -74,6 +74,53 @@
...
+
+    for (GoTypeDeclaration typeDeclaration : PsiTreeUtil.findChildrenOfType(file, GoTypeDeclaration.class)) {
+      List<GoTypeSpec> specList = typeDeclaration.getTypeSpecList();
+      if (typeDeclaration.getLparen() == null ||
+          typeDeclaration.getRparen() == null) {
+        continue;
+      }
+      if (specList.size() < 2) {
+        continue;
+      }
```
Comment (ignatov) [inline src/com/goide/editor/GoFoldingBuilder.java]: The same condition, please extract.

File: src/com/goide/editor/GoFoldingBuilder.java [hunk 2]
```diff
@@ -74,6 +74,53 @@
       foldTypes(type.getSpecType().getType(), result);
     }
 
+    for (GoVarDeclaration varDeclaration : PsiTreeUtil.findChildrenOfType(file, GoVarDeclaration.class)) {
+      List<GoVarSpec> specList = varDeclaration.getVarSpecList();
+      if (varDeclaration.getLparen() == null ||
+          varDeclaration.getRparen() == null) {
+        continue;
+      }
+      if (specList.size() < 2) {
```
Comment (ignatov) [inline src/com/goide/editor/GoFoldingBuilder.java]: You can combine these branches.

**Comment by dlsniper** [comment]
Cleaned up and ready for review. Thanks.

File: tests/com/goide/editor/GoFoldingBuilderTest.java [hunk 3]
```diff
@@ -34,4 +34,7 @@ protected String getBasePath() {
   public void testEmptyImportList()                               { doTest(); }
   public void testImportListWithNewLineAfterKeyword()             { doTest(); }
   public void testImportListWithOnlyThreeSymbolsToFold()          { doTest(); }
+  public void testVarDeclaration() { doTest(); }
```
Comment (ignatov) [inline tests/com/goide/editor/GoFoldingBuilderTest.java]: Please align the braces.

Comment (dlsniper) [inline tests/com/goide/editor/GoFoldingBuilderTest.java]: Done

**Comment by dlsniper** [comment]
@ignatov any other changes that I should do for the PR? Thank you.

**Comment by ignatov** [comment]
Merged manually, thanks

**Pull request closed without merging**